File: app/src/main/java/com/doordash/binding/viewmodel/RestaurantsModel.java
package com.doordash.binding.viewmodel;

import androidx.databinding.BaseObservable;
import androidx.databinding.Bindable;
import androidx.databinding.ObservableArrayList;


import java.util.List;

public class RestaurantsModel extends BaseObservable {

    @Bindable
    public  ObservableArrayList<StoreModel> restaurants;

    private volatile  static RestaurantsModel instance;
    public static RestaurantsModel getInstance(){
        if(null== instance){
            synchronized (RestaurantsModel.class) {
                if(null == instance) {
                    instance = new RestaurantsModel();
                    instance.restaurants = new ObservableArrayList<>();
                }
            }
        }
        return instance;
    }
    private RestaurantsModel(){

    }

//    public RestaurantsModel(ObservableArrayList<StoreModel> restaurants) {
//        if(null == restaurants)
//        this.restaurants = restaurants;
//    }

    public void addRestaurants(List<StoreModel> list){
        restaurants.addAll(list);
    }
    public void clear(){
        restaurants.clear();
    }

    public int size(){
        return restaurants.size();
    }


}

File: app/src/main/java/com/doordash/binding/viewmodel/RestaurantsModel.kt
package com.doordash.binding.viewmodel

import androidx.databinding.BaseObservable
import androidx.databinding.Bindable
import androidx.databinding.ObservableArrayList
import androidx.lifecycle.MutableLiveData
import com.doordash.bean.Store

class RestaurantsModel private constructor() : BaseObservable() {

    var restaurants: ObservableArrayList<Store> = ObservableArrayList()////Bindable should move to get, works with   notifyPropertyChanged()

    fun addRestaurants(list: List<Store>) {
        restaurants.addAll(list)
    }

    fun clear() {
        restaurants.clear()
    }

    fun size(): Int {
        return restaurants.size
    }

    companion object {
        val instance:RestaurantsModel by lazy (mode= LazyThreadSafetyMode.SYNCHRONIZED ){
            RestaurantsModel();
        }
    }
}